staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -206,6 +211,31 @@ func GetHTTPClient(req *http.Request) string {
 	return "unknown"
 }
 
+// MaybeTruncateUserAgent truncate User-Agent if too long, otherwise return it directly.
+func MaybeTruncateUserAgent(req *http.Request) string {
+	ua := GetHTTPClient(req)
+	if len(ua) > maxUserAgentLength {
+		ua = ua[:maxUserAgentLength] + userAgentTruncateSuffix
+	}
+
+	return ua
+}
+
+// LazyClientIp implements String() string and it will
+// calls GetClientIP() lazily only when required.
+type LazyClientIp struct {
```

Comment:
```suggestion
type LazyClientIP struct {
```

Reply:
similarly, this doesn't seem general purpose enough to include in the utils package, I'd make private to the handlers package for now